test/unit/ubuntu_test.rb

```diff
@@ -0,0 +1,45 @@
+require 'test_helper'
+
+class UbuntuTest < ActiveSupport::TestCase
+
+  test "os label for ubuntu" do
+    assert_equal "Ubuntu 14.10", operatingsystems(:ubuntu14_10).to_label
+  end
+
+  test "kernel location for 64bit arch ubuntu" do
+    host = FactoryGirl.create(:host, :operatingsystem => operatingsystems(:ubuntu14_10),
+                              :architecture => architectures(:x86_64))
+    assert_equal "boot/Ubuntu-14.10-x86_64-linux", host.os.kernel(host.arch)
+  end
+
+  test "initrd location for 64bit arch ubuntu" do
+    host = FactoryGirl.create(:host, :operatingsystem => operatingsystems(:ubuntu14_10),
+                              :architecture => architectures(:x86_64))
+    assert_equal "boot/Ubuntu-14.10-x86_64-initrd.gz", host.os.initrd(host.arch)
+  end
+
+  test "pxe prefix for ubuntu" do
+    host = FactoryGirl.create(:host, :operatingsystem => operatingsystems(:ubuntu14_10),
+                              :architecture => architectures(:x86_64))
+    prefix = host.os.pxe_prefix(host.arch)
+    assert_equal "boot/Ubuntu-14.10-x86_64", prefix
+  end
+
+  test "pxe files for ubuntu" do
+    host = FactoryGirl.create(:host, :with_medium,
+                              :operatingsystem => operatingsystems(:ubuntu14_10),
+                              :architecture => architectures(:x86_64),
+                              :medium => media(:ubuntu) )
+    host.medium.operatingsystems << host.os
+    host.arch.operatingsystems << host.os
+
+    prefix = host.os.pxe_prefix(host.arch).to_sym
+    pxe_files = host.os.pxe_files(host.medium, host.arch)
+
+    kernel = { prefix => "http://sg.archive.ubuntu.com/dists/utopic/main/installer-amd64/current/images/netboot/ubuntu-installer/amd64/linux"  }
+    initrd = { prefix => "http://sg.archive.ubuntu.com/dists/utopic/main/installer-amd64/current/images/netboot/ubuntu-installer/amd64/initrd.gz" }
+    assert pxe_files.include?(kernel)
+    assert pxe_files.include?(initrd)
+  end
```

Comment:
These tests are identical to the ones in ```coreos_test.rb``` and ```debian_test.rb```. Could you create some sort of template for these and just pass as arguments the things that change? It looks like the only difference between these tests is the word "ubuntu" and the kernel, initrd urls.